src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -817,19 +823,24 @@ def _visit_min_over(self, node: foast.Call, **kwargs) -> foast.Call:
         return self._visit_reduction(node, **kwargs)
 
     def _visit_astype(self, node: foast.Call, **kwargs) -> foast.Call:
-        casted_obj_type = node.args[0].type
-        dtype_obj = node.args[1]
-        assert isinstance(dtype_obj, foast.Name)
-        dtype_obj_type = dtype_obj.type
-        assert isinstance(dtype_obj_type, ts.FunctionType)
-        assert dtype_obj.id in fbuiltins.TYPE_BUILTIN_NAMES
-        assert isinstance(casted_obj_type, ts.FieldType)
-        assert type_info.is_arithmetic(casted_obj_type) or type_info.is_logical(casted_obj_type)
+        value, new_type = node.args
```

Comment:
can we add a warning in case someone uses astype to cast the same type?

Reply:
As discussed this is actually a feature for variable precision.